services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java

```diff
@@ -21,19 +23,14 @@
 public class ProdPpacIosRateLimitStrategy implements PpacIosRateLimitStrategy {
 
   private static final Logger logger = LoggerFactory.getLogger(ProdPpacIosRateLimitStrategy.class);
-  static final int VALIDITY_IN_HOURS = 23;
 
-  private final Clock clock;
+  private final int validityInSeconds;
 
-  @Autowired
-  public ProdPpacIosRateLimitStrategy() {
-    this.clock = Clock.systemUTC();
-  }
-
-  ProdPpacIosRateLimitStrategy(Clock clock) {
-    this.clock = clock;
-    logger.warn("Was started with clock {}. Constructor is intended for testing purposes. DO NO USE IN PRODUCTION!",
-        clock);
+  /**
+   * Constructs a validator instance.
+   */
+  public ProdPpacIosRateLimitStrategy(PpacConfiguration ppacConfiguration) {
```

Comment:
Maybe we can keep the configuration as a field instead of the integer itself. So we have more flexibility when we need more/different configuration. Also makes testing easier :) What do you think?

Reply:
We can have it either way. My rationale here was that it makes it easier to understand and immediately see what is used from the configuration. Why would it make testing easier?

Reply:
Agree but I had the feeling that when I was testing this locally, either Spring and Mockito had trouble injecting a Mock when the Constructor Parameter is not assigned to a field.

Reply:
I solved this with a much simplified test, not even using Mockito and Spring...